File: baekjoon/5582.cpp
#include <iostream>
#include <vector>
#include <string>
#include <limits.h>

using namespace std;

int maxValue = 0;

int main()
{
    string s1, s2;
    cin >> s1 >> s2;

    vector<vector<int> > lcs(s2.size() + 1, vector<int>(s1.size() + 1, 0));
    for (int i = 0; i < s2.size() + 1; i++)
    {
        for (int j = 0; j < s1.size() + 1; j++)
        {
            if (i == 0 || j == 0)
                continue;
            if (s2[i - 1] == s1[j - 1])
            {
                lcs[i][j] = lcs[i - 1][j - 1] + 1;
                maxValue = max(maxValue, lcs[i][j]);
            }
        }
    }

    cout << maxValue;
}

File: baekjoon/5582.py
str1 = input().strip()
str2 = input().strip()

table = [[0] * (len(str2) + 1) for _ in range(len(str1) + 1)]

max_value = 0
for i in range(1, len(str1) + 1):
    for j in range(1, len(str2) + 1):
        if (str1[i - 1] == str2[j - 1]):
            table[i][j] = table[i - 1][j - 1] + 1
            max_value = max(table[i][j], max_value)
print(max_value)
